extensions/adapters/raster/src/main/java/org/locationtech/geowave/adapter/raster/operations/DeletePyramidLevelCommand.java

```diff
@@ -0,0 +1,222 @@
+        }
+      }
+      if (found) {
+        break;
+      }
+
+    }
+    if (!found) {
+      LOGGER.error("Store has no indices supporting pyramids.");
+      return;
+    }
+    // delete the resolution from the overview, delete the partitions, and delete the data
+    store.queryStatistics(
+        OverviewStatistics.STATS_TYPE.newBuilder().dataType(adapter.getTypeName()).build());
+    final DataStatisticsStore statsStore = inputStoreLoader.createDataStatisticsStore();
+    final InternalAdapterStore adapterIdStore = inputStoreLoader.createInternalAdapterStore();
+    OverviewStatistics ovStats = null;
+    PartitionStatistics<?> pStats = null;
+    try (CloseableIterator<InternalDataStatistics<?, ?, ?>> it =
+        statsStore.getDataStatistics(
+            adapterIdStore.getAdapterId(adapter.getTypeName()),
+            OverviewStatistics.STATS_TYPE)) {
+      while (it.hasNext()) {
+        final InternalDataStatistics<?, ?, ?> next = it.next();
+        if (next instanceof OverviewStatistics) {
+          ovStats = (OverviewStatistics) next;
+          break;
+        }
+      }
+    }
+    if (ovStats == null) {
+      LOGGER.error("Unable to find overview stats for coverage " + adapter.getTypeName());
+      return;
+    }
+    if (!ovStats.removeResolution(res)) {
+      LOGGER.error("Unable to remove resolution for pyramid level " + level);
+      return;
+    }
+    try (CloseableIterator<InternalDataStatistics<?, ?, ?>> it =
+        statsStore.getDataStatistics(
+            adapterIdStore.getAdapterId(adapter.getTypeName()),
+            i.getName(),
+            PartitionStatistics.STATS_TYPE)) {
+      while (it.hasNext()) {
+        final InternalDataStatistics<?, ?, ?> next = it.next();
+        if (next instanceof PartitionStatistics) {
+          pStats = (PartitionStatistics) next;
+          break;
+        }
+      }
+    }
+    final byte[][] predefinedSplits = i.getIndexStrategy().getPredefinedSplits();
+    // this should account for hash partitioning if used
+    final List<ByteArray> partitions = new ArrayList<>();
+    if ((predefinedSplits != null) && (predefinedSplits.length > 0)) {
+      for (final byte[] split : predefinedSplits) {
+        partitions.add(new ByteArray(Arrays.append(split, level.byteValue())));
+      }
+    } else {
+      partitions.add(new ByteArray(new byte[] {level.byteValue()}));
+    }
+    if (pStats == null) {
+      LOGGER.error(
+          "Unable to find partition stats for coverage "
+              + adapter.getTypeName()
+              + " and index "
+              + i.getName());
+      return;
+    }
+    for (final ByteArray p : partitions) {
+      if (!pStats.getPartitionKeys().remove(p)) {
+        LOGGER.error(
+            "Unable to remove partition " + p.getHexString() + " for pyramid level " + level);
+        return;
+      }
+    }
+    statsStore.setStatistics(ovStats);
+    statsStore.setStatistics(pStats);
+
+    for (final ByteArray p : partitions) {
```

Comment:
Is it possible that there could be no statistics for the given coverage, but still have data that could be removed?

Reply:
yeah, I think the only somewhat legit case may be is statistics are disabled by the options, and I added a check for that (although with raster data functionality would be greatly diminished if stats are disabled so its probably something that just shouldn't happen, the overview stat is used to pick a resolution if you want to composite an image based on geobounds and pixel height/width, without that you would primarily be needing to query for tiles and composite them into an image externally).

Otherwise if there stats aren't found, something is wrong and it stands to reason you don't want to delete the data without properly maintaining the stats and get in a more chaotic situation.  The only thing I can think of is to add a `--force` flag that does it anyways, but probably outside the scope here.